backend/entities/oauth/base/controller.go

```diff
@@ -0,0 +1,92 @@
+package base
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/GenerateNU/sac/backend/entities/models"
+	"github.com/GenerateNU/sac/backend/utilities"
+	"github.com/gofiber/fiber/v2"
+)
+
+type OAuthController struct {
+	OAuthService OAuthServiceInterface
+}
+
+func NewOAuthController(oauthService OAuthServiceInterface) *OAuthController {
+	return &OAuthController{OAuthService: oauthService}
+}
+
+func (oc *OAuthController) Authorize(c *fiber.Ctx) error {
+	// Extract the resource type from the query params:
+	resourceType := models.OAuthResource(c.Query("type"))
+	if resourceType == "" {
+		return errors.New("resource type is required")
+	}
+
+	// Extract the user making the call:
+	userID := c.Locals("userID").(string)
```

Comment:
we should use a private type as the key to avoid any potential collisions

Reply:
also, isn't this stored in the custom claims as the issuer? it is set [here](https://github.com/GenerateNU/sac/blob/main/backend/entities/auth/base/service.go#L71) when logging in